index.html

```diff
@@ -456,105 +464,107 @@ <h2>
+          </li>
+          <li>
+            Let |record| be the <a>platform wake lock</a>'s
+            <a>state record</a> associated with
+            <code>this</code>'s |type|.
+          </li>
+          <li>
+            If |signal|’s <a>aborted flag</a> is set, then
+            reject |promise| with an "<a>AbortError</a>"
+            <a>DOMException</a>, and abort these steps.
+          </li>
+          <li>
+            If |signal| is not <code>null</code>, then
+            <a>add the following abort steps</a> to |signal|:
+            <ol>
+              <li>
+                Decrease |record|.<a>[[\RequestCounter]]</a> by one.
+              </li>
+              <li>
+                If |record|.<a>[[\RequestCounter]]</a> is 0, then
+                run the following steps <a>in parallel</a>:
+                <ol>
+                  <li>
+                    Run <a>release a wake lock</a> on <code>this</code>.
+                  </li>
+                </ol>
+              </li>
+              <li>
+                Reject |promise| with an "<a>AbortError</a>"
+                <a>DOMException</a>, and abort these steps.
+              </li>
+            </ol>
+          </li>
+          <li>
+            If the <a>current global object</a> is not a <a>DedicatedWorkerGlobalScope</a>
+            object, run the following steps:
+            <ol>
+              <li>
+                Let |document| be the <a>responsible document</a> of the
+                <a>current settings object</a>.
+              </li>
+              <li>
+                If |document| is not <a>fully active</a>,
+                reject |promise| with a <code>"<a>NotAllowedError</a>"</code>
+                <a>DOMException</a>, and abort these steps.
+              </li>
+              <li>
+                If <code>this</code>'s |type| is
+                "<a data-lt="WakeLockType.screen">screen</a>" and the
+                <a>Document</a> of the <a>top-level browsing context</a>
+                is <a>hidden</a>, reject |promise|
+                with a <code>"<a>NotAllowedError</a>"</code>
+                <a>DOMException</a>, and abort these steps.
+              </li>
+            </ol>
+          </li>
+          <li>
+            Increase |record|.<a>[[\RequestCounter]]</a> by one.
+          </li>
+          <li>
+            If |record|.<a>[[\RequestCounter]]</a> is 1,
+            run the following steps <a>in parallel</a>:
+            <ol>
+              <li>
+                Let |success| be the the result of running
+                <a>acquire a wake lock</a> on <code>this</code>.
+              </li>
+              <li>
+                If |success| is <code>false</code> then reject
+                |promise| with a <code>"<a>NotAllowedError</a>"</code>
+                <a>DOMException</a>, and abort these steps.
+              </li>
+            </ol>
+          </li>
+          <li>
+            Fulfill |promise|.
```

Comment:
```suggestion
            Fulfill |promise| with ???
```

Reply:
Missing the type above.

Reply:
with `undefined`?

Reply:
It's weird to say "Fulfill" ... a promise is fulfilled by either rejecting with error or resolving (with a var or value).

Reply:
Ah so just write `Resolve |promise|` ? or do you need to say `with ...`

Reply:
Resolve |promise| with <code>void</code>.

(which resolves it with undefined in JS land)

Reply:
Same with other places in the spec.